lib/TokenManager.ts

```diff
@@ -100,7 +108,7 @@ function setExpireEventTimeoutAll(sdk, tokenMgmtRef, storage) {
   }
 }
 
-function add(sdk, tokenMgmtRef, storage, key, token: Token) {
+function add(sdk, tokenMgmtRef, storage, key, token: Token, shouldEmitEvent = true) {
```

Comment:
I'm not a big fan of adding boolean flags to function signatures unless we're passing an object.  It makes the code harder to skim (as you prove below where you add a comment to say what "false" means).  

If these are all internal methods, can we switch this to pass an object and thus behave as named parameters?

Reply:
This one actually is a public method (https://github.com/okta/okta-auth-js/blob/master/lib/TokenManager.ts#L281)